src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -43225,6 +43222,106 @@ static void F4(object? x4)
                 Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "f()").WithLocation(37, 25));
         }
 
+        [Fact]
+        [WorkItem(29892, "https://github.com/dotnet/roslyn/issues/29892")]
+        public void LocalFunction_02()
+        {
+            var source =
+@"class C
+{
+    static void F1()
+    {
+        string? x = """";
+        f();
+        x = """";
+        g();
+        void f()
+        {
+            x.ToString();
+            x = null;
+            f();
+        }
+        void g()
+        {
+            x.ToString();
+            x = null;
+        }
+    }
+}";
+            var comp = CreateCompilation(new[] { source }, options: WithNonNullTypesTrue());
+            comp.VerifyDiagnostics(
+                // (11,13): warning CS8602: Dereference of a possibly null reference.
+                //             x.ToString();
```

Comment:
Please put a comment on the appropriate line. I have no idea which `x.ToString()` is causing this.

Reply:
And on the other tests.

---
In reply to: [358966254](https://github.com/dotnet/roslyn/pull/40422#discussion_r358966254) [](ancestors = 358966254)